azure-iot-device/azure/iot/device/common/pipeline/pipeline_ops_http.py

```diff
@@ -0,0 +1,63 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from . import PipelineOperation
+
+
+class SetHTTPConnectionArgsOperation(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to connect to a server using the HTTP protocol.
+
+    This operation is in the group of HTTP operations because its attributes are very specific to the HTTP protocol.
+    """
+
+    def __init__(self, hostname, callback, ca_cert=None, client_cert=None, sas_token=None):
+        """
+        Initializer for SetHTTPConnectionArgsOperation objects.
+        :param str hostname: The hostname of the HTTP server we will eventually connect to
+        :param str ca_cert: (Optional) The CA certificate to use if the HTTP server that we're going to
+            connect to uses server-side TLS
+        :param X509 client_cert: (Optional) The x509 object containing a client certificate and key used to connect
+            to the HTTP service
+        :param str sas_token: The token string which will be used to authenticate with the service
+        :param Function callback: The function that gets called when this operation is complete or has failed.
+            The callback function must accept A PipelineOperation object which indicates the specific operation which
+            has completed or failed.
+        """
+        super(SetHTTPConnectionArgsOperation, self).__init__(callback=callback)
+        self.hostname = hostname
+        self.ca_cert = ca_cert
+        self.client_cert = client_cert
+        self.sas_token = sas_token
+
+
+class HTTPRequestAndResponseOperation(PipelineOperation):
```

Comment:
As we've discussed offline, there should be a way to restructure this pipeline in order to not need this stage. The generic RequestAndResponseOperation from pipeline_stages_base should be sufficient, and if it's not, we need to get to the bottom of why.